server.go

```diff
@@ -24,113 +26,140 @@ type Route struct {
 	Params map[string]string `yaml:"-"`
 }
 
+// Routes is a list of Routes.
 type Routes struct {
 	Routes []Route `yaml:"routes"`
 }
 
-type ByPathLength []Route
+type byPathLength []Route
 
-func (b ByPathLength) Len() int           { return len(b) }
-func (b ByPathLength) Swap(i, j int)      { b[i], b[j] = b[j], b[i] }
-func (b ByPathLength) Less(i, j int) bool { return len(b[i].Path) > len(b[j].Path) }
+func (b byPathLength) Len() int           { return len(b) }
+func (b byPathLength) Swap(i, j int)      { b[i], b[j] = b[j], b[i] }
+func (b byPathLength) Less(i, j int) bool { return len(b[i].Path) > len(b[j].Path) }
 
-const TIMEOUT_MILLISECONDS = 500
+const timeout = 500 * time.Millisecond
 
-var ErrInternalServerGeneric = errors.New("oops, something went wrong")
+var errInternalServerGeneric = errors.New("oops, something went wrong")
 
 func main() {
+	if err := run(); err != nil {
+		log.Fatalf("Error: %v\n", err)
+	}
+}
 
-	cgi_scripts_dir := os.Getenv("CGI_DIR")
-	if cgi_scripts_dir == "" {
-		log.Fatalf("ERROR: CGI_DIR environment variable not set\n")
+func run() error {
```

Comment:
The go error handling is a bit strange. One idea is, to return the error as far as possible and only log it at one place. A `run() error` function helps with that.